### PR TITLE
chore: adds rule to require design review on branches with styling/front end changes

### PR DESCRIPTION
**Pull request by shkeating**
# SC-1209

adds a rule to prompt design review on any PR that has front end changes, by checking for updates in the `stories` directory or updates to any `.stories.tsx` files. uses the [Orbit design team group](https://github.com/orgs/USSF-ORBIT/teams/design) for this.

## Reviewer notes

I'd like to try this so I can know what branches I can help with my reviewing, if we decide it's annoying or a bottleneck, we can always undo it. 

Should I be creating a decision record for this somewhere?

I followed the pattern in MilMove's repo for this. Is there any other files I need to update for this to work?

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -12,5 +17,5 @@
 
 # Solution: make files used by renovate owned by everyone
 
-**/*/package.json
-**/*/yarn.lock
+**/\*/package.json
+**/\*/yarn.lock
```
Comment (gidjin) [inline .github/CODEOWNERS:21]: Suggestion: remove the backslash. I think this was likely added by your editor.

**Review by gidjin — COMMENTED** [review]
We don't have any github groups, so we might have to manually specify everyone like we do the default code owners, but maybe Minh or Barry have a better suggestion :)

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1,6 +1,11 @@
 # Default owners for everything
 
-* @abbyoung @esacteksab @gidjin @minhlai
+- @abbyoung @esacteksab @gidjin @minhlai
+
+# Require design / frontend reviews on Storybook
+
+/src/stories/ @USSF-ORBIT/design
+\*.stories.tsx @USSF-ORBIT/design
```
[the remark this answers is not in the document]
Comment (esacteksab) [reply .github/CODEOWNERS:8]: I'll yield to @minhlai since he's lead.

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -12,5 +17,5 @@
 
 # Solution: make files used by renovate owned by everyone
 
-**/*/package.json
-**/*/yarn.lock
+**/\*/package.json
```
Comment (esacteksab) [inline .github/CODEOWNERS:20]:
```suggestion
**/*/package.json
```

File: .github/CODEOWNERS (hunk 1 above)
Comment (esacteksab) [inline .github/CODEOWNERS:21]:
```suggestion
**/*/yarn.lock
```

File: .github/CODEOWNERS (hunk 2 above)
Comment (esacteksab) [inline .github/CODEOWNERS:8]:
```suggestion
*.stories.tsx @USSF-ORBIT/design
```

[the remark this answers is not in the document]
Comment (gidjin) [reply .github/CODEOWNERS:8]: I was also under the impression that the group didn't exist which was wrong. However, how we maintain it is still a question that interests me.

**Review by esacteksab — APPROVED** [review]

**Pull request merged**